### PR TITLE
PM-10617 + PM-11270 fixes after QA review

### PR DESCRIPTION
**Pull request by dseverns-livefront**
## 🎟️ Tracking
https://bitwarden.atlassian.net/browse/PM-10617
https://bitwarden.atlassian.net/browse/PM-11270

## 📔 Objective
- fix discrepancies in some of the copy shown on the new and "legacy" create account screens

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/setpassword/SetPasswordScreen.kt [hunk 1]
```diff
@@ -170,7 +170,7 @@ private fun SetPasswordScreenContent(
         Spacer(modifier = Modifier.height(16.dp))
 
         BitwardenTextField(
-            label = stringResource(id = R.string.master_password_hint),
+            label = stringResource(id = R.string.master_password_hint_optional),
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/setpassword/SetPasswordScreen.kt:173]: Why change the name?

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/completeregistration/CompleteRegistrationScreen.kt [hunk 2]
```diff
@@ -236,11 +236,15 @@ private fun CompleteRegistrationContent(
 
         var showPassword by rememberSaveable { mutableStateOf(false) }
         BitwardenPasswordField(
-            label = stringResource(id = R.string.master_password),
+            label = stringResource(id = R.string.master_password_required)
+                .takeIf { showNewOnboardingUi }
+                ?: stringResource(id = R.string.master_password),
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/completeregistration/CompleteRegistrationScreen.kt:241]: Could we do this instead?
```kotlin
label = stringResource(
    id = R.string.master_password_required
        .takeIf { showNewOnboardingUi }
        ?: R.string.master_password,
)
```

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/setpassword/SetPasswordScreen.kt (hunk 1 above)
Comment (david-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/setpassword/SetPasswordScreen.kt:173]: Hmmm, I see why you did it now.

@SaintPatrck is this going to confuse Crowdin?

Comment (dseverns-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/setpassword/SetPasswordScreen.kt:173]: I can make the new one "m_p_h_not_optional" haha

Comment (dseverns-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/setpassword/SetPasswordScreen.kt:173]: not_specified *

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/completeregistration/CompleteRegistrationScreen.kt (hunk 2 above)
Comment (dseverns-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/completeregistration/CompleteRegistrationScreen.kt:241]: making this change throughout the file for similar places

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/completeregistration/CompleteRegistrationScreen.kt [hunk 3]
```diff
@@ -141,11 +141,9 @@ fun CompleteRegistrationScreen(
...
         topBar = {
             BitwardenTopAppBar(
-                title = if (state.onboardingEnabled) {
-                    stringResource(id = R.string.create_account)
-                } else {
-                    stringResource(id = R.string.set_password)
-                },
+                title = stringResource(id = R.string.create_account
+                    .takeIf { state.onboardingEnabled }
+                    ?: R.string.set_password),
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/completeregistration/CompleteRegistrationScreen.kt:146]: Formatting:

```kotlin
stringResource(
    id = R.string.create_account
        .takeIf { state.onboardingEnabled }
        ?: R.string.set_password,
),
```

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/setpassword/SetPasswordScreen.kt (hunk 1 above)
Comment (david-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/setpassword/SetPasswordScreen.kt:173]: I think that would work

Comment (SaintPatrck) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/setpassword/SetPasswordScreen.kt]: > Hmmm, I see why you did it now.
> 
> @SaintPatrck is this going to confuse Crowdin?

I see you reverted renaming that resource so it's moot now, but for future reference, renaming will result in loss of existing translations for the resource being renamed.

**Review by SaintPatrck — APPROVED** [review]

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/setpassword/SetPasswordScreen.kt (hunk 1 above)
Comment (david-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/setpassword/SetPasswordScreen.kt]: OK, thanks @SaintPatrck

**Review by david-livefront — APPROVED** [review]

**Pull request merged**